app/controllers/podcast_metrics_controller.rb

```diff
@@ -96,17 +224,84 @@ def metrics_params
       )
   end
 
+  def uniques_params
+    params
+      .permit(:uniques_selection)
+      .with_defaults(
+        uniques_selection: "last_7_rolling"
+      )
+      .merge(metrics_params)
+  end
+
+  def dropdays_params
+    params
+      .permit(:dropday_range, :interval)
+      .with_defaults(
+        dropday_range: 7
+      )
+      .merge(metrics_params)
+  end
+
   def episode_rollups(episodes, rollups, totals)
-    episodes.map do |ep|
+    episodes.to_enum(:each_with_index).map do |episode, i|
+      {
+        episode: episode,
+        rollups: rollups.select do |r|
+          r["episode_id"] == episode.guid
+        end,
+        totals: totals.select do |r|
+          r["episode_id"] == episode.guid
+        end,
+        color: colors[i]
+      }
+    end
+  end
+
+  def episode_dropdays(episodes, rollups, totals)
+    episodes.to_enum(:each_with_index).map do |episode, i|
       {
-        ep: ep,
+        episode: episode,
         rollups: rollups.select do |r|
-          r["episode_id"] == ep.guid
+          r["episode_id"] == episode.guid
         end,
         totals: totals.select do |r|
-          r["episode_id"] == ep.guid
-        end
+          r["episode_id"] == episode.guid
+        end,
+        color: colors[i]
       }
     end
   end
+
+  def generate_date_range(date_start, date_end, interval)
+    start_range = date_start.to_datetime.utc.send(:"beginning_of_#{interval.downcase}")
+    end_range = date_end.to_datetime.utc.send(:"beginning_of_#{interval.downcase}")
+    range = []
+    i = 0
+
+    while start_range + i.send(:"#{interval.downcase.pluralize}") <= end_range
+      range << start_range + i.send(:"#{interval.downcase.pluralize}")
+      i += 1
+    end
```

Comment:
While loops do sometimes make me nervous.  But, this seems to work pretty well.

If this does became slow someday, I'd probably look into something more like the [DateTime.step](https://ruby-doc.org/stdlib-2.6.3/libdoc/date/rdoc/Date.html#method-i-step) method.  But, probably need more than a few hundred to need any optimizing.